src/components/safe-apps/AddCustomAppModal/index.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback } from 'react'
+import type { SubmitHandler } from 'react-hook-form'
+import { useForm } from 'react-hook-form'
+import {
+  DialogActions,
+  DialogContent,
+  Typography,
+  Button,
+  TextField,
+  FormControlLabel,
+  Checkbox,
+  Link,
+  Box,
+  FormHelperText,
+} from '@mui/material'
+import CheckIcon from '@mui/icons-material/Check'
+import type { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined'
+import ModalDialog from '@/components/common/ModalDialog'
+import { isValidURL } from '@/utils/validation'
+import { useCurrentChain } from '@/hooks/useChains'
+import useChainId from '@/hooks/useChainId'
+import useAsync from '@/hooks/useAsync'
+import useDebounce from '@/hooks/useDebounce'
+import { fetchSafeAppFromManifest } from '@/services/safe-apps/manifest'
+import { SAFE_APPS_EVENTS, trackEvent } from '@/services/analytics'
+import { trimTrailingSlash, isSameUrl } from '@/utils/url'
+import { AppRoutes } from '@/config/routes'
+import CustomAppPlaceholder from './CustomAppPlaceholder'
+import CustomApp from './CustomApp'
+
+import css from './styles.module.css'
+
+type Props = {
+  open: boolean
+  onClose: () => void
+  onSave: (data: SafeAppData) => void
+  // A list of safe apps to check if the app is already there
+  safeAppsList: SafeAppData[]
+}
+
+type CustomAppFormData = {
+  appUrl: string
+  riskAcknowledgement: boolean
+  safeApp: SafeAppData
+}
+
+const HELP_LINK = 'https://docs.gnosis-safe.io/build/sdks/safe-apps'
+
+const AddCustomAppModal = ({ open, onClose, onSave, safeAppsList }: Props) => {
+  const chainId = useChainId()
+  const currentChain = useCurrentChain()
```

Comment:
does `currentChain` contain `chainId`? if yes, you can get it from there instead calling `useChainId`